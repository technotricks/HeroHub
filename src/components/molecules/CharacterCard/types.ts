import {ItemClickCallback} from '@/types/base';
import {Character} from '@/types/character';

export type CardProps = {
  item: Character;
  onPress: ItemClickCallback;
};
