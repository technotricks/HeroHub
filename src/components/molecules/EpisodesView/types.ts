import {CharacterDetail} from '@/types/character';

export type EpisodesViewProps = {
  item?: CharacterDetail;
};
