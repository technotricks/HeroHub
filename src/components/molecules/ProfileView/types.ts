import {CharacterDetail} from '@/types/character';

export type ProfileViewProps = {
  item?: CharacterDetail;
};
