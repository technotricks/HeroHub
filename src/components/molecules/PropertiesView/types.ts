import {CharacterDetail} from '@/types/character';

export type PropertiesViewProps = {
  item?: CharacterDetail;
};
