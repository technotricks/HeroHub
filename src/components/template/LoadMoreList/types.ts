import {ItemClickCallback, LoadMoreCallback} from '@/types/base';
import {Character} from '@/types/character';

export namespace ILoadMore {
  export type Props = {
    items: Character[];
    isLoading: Boolean;
    onPress: ItemClickCallback;
    handleLoadMore: LoadMoreCallback;
  };
}
