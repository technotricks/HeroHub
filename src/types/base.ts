import {Character} from '@/types/character';

export type ItemClickCallback = (item: Character) => void;
export type LoadMoreCallback = () => void;
