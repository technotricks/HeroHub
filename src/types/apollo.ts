export type QueryResult<T> = {
  loading: boolean;
  data: T | undefined;
  error: Error | undefined;
};

export type Variables = {
  [key: string]: any;
};

export type QueryHookOptions = {
  fetchPolicy?:
    | 'cache-first'
    | 'cache-and-network'
    | 'network-only'
    | 'cache-only';
  variables?: Variables;
};
