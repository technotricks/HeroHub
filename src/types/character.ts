export interface Character {
  id: number;
  name: string;
  status: string;
  image: string;
  gender: string;
  type: string;
}

type NameObject = {
  name: string;
};

type Episode = {
  name: string;
  episode: string;
};

export interface CharacterDetail extends Character {
  origin: NameObject;
  location: NameObject;
  episode: Episode[];
}

type Info = {
  id: number;
  name: number;
  status: number;
};

// Create a TypeScript type for the query result
export type GetCharactersData = {
  characters: {
    results: Character[];
    info: Info;
  };
};

// Create a TypeScript type for the query result
export type GetCharacterData = {
  characters: {
    results: CharacterDetail[];
  };
};

export type FetchCharacters = (page: number) => void;
export type FetchCharacter = (id: number) => void;
