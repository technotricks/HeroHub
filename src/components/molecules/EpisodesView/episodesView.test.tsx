import 'react-native';
import React from 'react';

import {it, expect, jest, describe} from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer, {create, act} from 'react-test-renderer';
import EpisodeView from '.';
import {CharacterDetail} from '@/types/character';

describe('Episode component should render correctly', () => {
  const characterDetail: CharacterDetail = {
    id: 1,
    image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
    name: 'Rick Sanchez',
    gender: 'Male',
    status: 'Alive',
    origin: {
      name: 'Earth (C-137)',
    },
    location: {
      name: 'Citadel of Ricks',
    },
    episode: [
      {
        episode: 'S01E01',
        name: 'Pilot',
      },
      {
        episode: 'S01E02',
        name: 'Lawnmower Dog',
      },
    ],
    type: '',
  };
  const characterDetailNoEpisodes: CharacterDetail = {
    id: 1,
    image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
    name: 'Rick Sanchez',
    gender: 'Male',
    status: 'Alive',
    origin: {
      name: 'Earth (C-137)',
    },
    location: {
      name: 'Citadel of Ricks',
    },
    episode: [],
    type: '',
  };

  const view = create(<EpisodeView item={characterDetail} />);
  const noEpisodesView = create(
    <EpisodeView item={characterDetailNoEpisodes} />,
  );

  it('snapshot', () => {
    expect(view).toMatchSnapshot();
  });

  it('snapshot empty eposodes', () => {
    expect(noEpisodesView).toMatchSnapshot();
  });
});
