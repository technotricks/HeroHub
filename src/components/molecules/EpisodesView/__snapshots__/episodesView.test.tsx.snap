// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episode component should render correctly snapshot 1`] = `
[
  <View
    style={
      {
        "borderBottomColor": "#ccc",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "transparent",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "textAlign": "center",
        }
      }
    >
      <Text
        aligin="center"
        color="#0C0D34"
        fontSize="20px"
        isBold={true}
        style={
          {
            "color": "#0C0D34",
            "fontSize": 20,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Episodes
      </Text>
    </View>
  </View>,
  <View
    bgColor="#E8FFFA"
    style={
      {
        "backgroundColor": "#E8FFFA",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "transparent",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "textAlign": "center",
        }
      }
    >
      <Text
        aligin="center"
        color="#0C0D34"
        fontSize="14px"
        style={
          {
            "color": "#0C0D34",
            "fontSize": 14,
            "fontWeight": "normal",
            "textAlign": "center",
          }
        }
      >
        S01E01
      </Text>
    </View>
    <View
      style={
        {
          "backgroundColor": "transparent",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "textAlign": "center",
        }
      }
    >
      <Text
        aligin="left"
        color="#0C0D34"
        fontSize="14px"
        style={
          {
            "color": "#0C0D34",
            "fontSize": 14,
            "fontWeight": "normal",
            "textAlign": "left",
          }
        }
      >
        Pilot
      </Text>
    </View>
  </View>,
  <View
    bgColor="#FFE4D9"
    style={
      {
        "backgroundColor": "#FFE4D9",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "transparent",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "textAlign": "center",
        }
      }
    >
      <Text
        aligin="center"
        color="#0C0D34"
        fontSize="14px"
        style={
          {
            "color": "#0C0D34",
            "fontSize": 14,
            "fontWeight": "normal",
            "textAlign": "center",
          }
        }
      >
        S01E02
      </Text>
    </View>
    <View
      style={
        {
          "backgroundColor": "transparent",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "textAlign": "center",
        }
      }
    >
      <Text
        aligin="left"
        color="#0C0D34"
        fontSize="14px"
        style={
          {
            "color": "#0C0D34",
            "fontSize": 14,
            "fontWeight": "normal",
            "textAlign": "left",
          }
        }
      >
        Lawnmower Dog
      </Text>
    </View>
  </View>,
]
`;

exports[`Episode component should render correctly snapshot empty eposodes 1`] = `
<View
  style={
    {
      "borderBottomColor": "#ccc",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "transparent",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "textAlign": "center",
      }
    }
  >
    <Text
      aligin="center"
      color="#0C0D34"
      fontSize="20px"
      isBold={true}
      style={
        {
          "color": "#0C0D34",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      No Episodes found
    </Text>
  </View>
</View>
`;
